packages/atomic/src/components/result-lists/atomic-result-list/atomic-result-list.tsx

```diff
@@ -0,0 +1,152 @@
+  public resultList!: ResultList;
+  public resultsPerPage!: ResultsPerPage;
+  public resultTemplatesManager!: ResultTemplatesManager<TemplateContent>;
+  public listWrapperRef?: HTMLDivElement;
+
+  /**
+   * TODO: KIT-452 Infinite scroll feature
+   * Whether to automatically retrieve an additional page of results and append it to the
+   * current results when the user scrolls down to the bottom of element
+   */
+  private enableInfiniteScroll = false;
+  private renderingFunction?: (result: Result) => HTMLElement = undefined;
+
+  @Element() public host!: HTMLDivElement;
+
+  @BindStateToController('resultList')
+  @State()
+  public resultListState!: ResultListState;
+
+  @BindStateToController('resultsPerPage')
+  @State()
+  public resultsPerPageState!: ResultsPerPageState;
+
+  @State() public ready = false;
+
+  @State() public error!: Error;
+  @State() public templateHasError = false;
+
+  /**
+   * A list of non-default fields to include in the query results, separated by commas.
+   * The default fields sent in a request are: 'date', 'author', 'source', 'language', 'filetype', 'parents', ‘urihash’, ‘objecttype’, ‘collection’, ‘permanentid’ 'ec_price', 'ec_name', 'ec_description', 'ec_brand', 'ec_category', 'ec_item_group_id', 'ec_shortdesc', 'ec_thumbnails', 'ec_images', 'ec_promo_price', 'ec_in_stock', 'ec_cogs', and 'ec_rating'.
+   */
+  @Prop({reflect: true}) public fieldsToInclude = '';
+  /**
+   * The desired layout to use when displaying results. Layouts affect how many results to display per row and how visually distinct they are from each other.
+   */
+  @Prop({reflect: true}) public display: ResultDisplayLayout = 'list';
+  /**
+   * The spacing of various elements in the result list, including the gap between results, the gap between parts of a result, and the font sizes of different parts in a result.
+   */
+  @Prop({reflect: true}) density: ResultDisplayDensity = 'normal';
+  /**
+   * The expected size of the image displayed in the results.
+   */
+  @Prop({reflect: true}) imageSize?: ResultDisplayImageSize;
+  /**
+   * @deprecated use `imageSize` instead.
+   */
+  @Prop({reflect: true}) image: ResultDisplayImageSize = 'icon';
+
+  /**
+   * Sets a rendering function to bypass the standard HTML template mechanism for rendering results.
+   * You can use this function while working with web frameworks that don't use plain HTML syntax, e.g., React, Angular or Vue.
+   *
+   * Do not use this method if you integrate Atomic in a plain HTML deployment.
+   *
+   * @param render
+   */
+  @Method() public async setRenderFunction(
+    render: (result: Result) => HTMLElement
+  ) {
+    this.renderingFunction = render;
+  }
+
+  @Listen('scroll', {target: 'window'})
+  handleInfiniteScroll() {
+    handleInfiniteScroll(this.enableInfiniteScroll, this.host, this.resultList);
+  }
+
+  public async initialize() {
+    if (this.host.innerHTML.includes('<atomic-result-children')) {
```

Comment:
```suggestion
    if (this.host.innerHTML.includes('<atomic-result-children>')) {
```
or without brackets

Or you could use `querySelector()` ?

Reply:
this.host.shadowRoot.querySelector should work

Reply:
the problem is that it's nested within atomic-result's shadow dom too. so I could do `this.host?.shadowRoot?.querySelector('atomic-result').shadowRoot.querySelector('atomic-result-children')` but it doesn't seem to have rendered anything at that point, so `shadowRoot?.querySelector('atomic-result')` is actually null